Lib/java/std_map.i

```diff
@@ -12,60 +14,188 @@
+
+%typemap(javabase) std::map<KeyType, ValueType, Comparator>
+    "java.util.AbstractMap<$typemap(jboxtype, KeyType), $typemap(jboxtype, ValueType)>"
+
+%proxycode %{
+
+  public int size() {
+    return sizeImpl();
+  }
+
+  public boolean containsKey(Object key) {
+    if (!(key instanceof $typemap(jboxtype, KeyType))) {
+      return false;
+    }
+
+    return containsImpl(($typemap(jboxtype, KeyType))key);
+  }
+
+  public $typemap(jboxtype, ValueType) get(Object key) {
+    if (!(key instanceof $typemap(jboxtype, KeyType))) {
+      return null;
+    }
+
+    Iterator itr = find(($typemap(jboxtype, KeyType)) key);
+    if (itr.isNot(end())) {
+      return itr.getValue();
+    }
+
+    return null;
+  }
+
+  public $typemap(jboxtype, ValueType) put($typemap(jboxtype, KeyType) key,
+                                           $typemap(jboxtype, ValueType) value) {
+    Iterator itr = find(($typemap(jboxtype, KeyType)) key);
+    if (itr.isNot(end())) {
+      $typemap(jboxtype, ValueType) oldValue = itr.getValue();
+      itr.setValue(value);
+      return oldValue;
+    } else {
+      putUnchecked(key, value);
+      return null;
+    }
+  }
+
+  public $typemap(jboxtype, ValueType) remove(Object key) {
+    if (!(key instanceof $typemap(jboxtype, KeyType))) {
+      return null;
+    }
+
+    Iterator itr = find(($typemap(jboxtype, KeyType)) key);
+    if (itr.isNot(end())) {
+      $typemap(jboxtype, ValueType) oldValue = itr.getValue();
+      removeUnchecked(itr);
+      return oldValue;
+    } else {
+      return null;
+    }
+  }
+
+  public java.util.Set<Entry<$typemap(jboxtype, KeyType), $typemap(jboxtype, ValueType)>> entrySet() {
+    java.util.Set<Entry<$typemap(jboxtype, KeyType), $typemap(jboxtype, ValueType)>> setToReturn =
+        new java.util.HashSet<Entry<$typemap(jboxtype, KeyType), $typemap(jboxtype, ValueType)>>();
+
+    Iterator itr = begin();
+    final Iterator end = end();
+    while(itr.isNot(end)) {
+      setToReturn.add(new Entry<$typemap(jboxtype, KeyType), $typemap(jboxtype, ValueType)>() {
+        private Iterator iterator;
+
+        private Entry<$typemap(jboxtype, KeyType), $typemap(jboxtype, ValueType)> init(Iterator iterator) {
+          this.iterator = iterator;
+          return this;
+        }
+
+        public $typemap(jboxtype, KeyType) getKey() {
+          return iterator.getKey();
+        }
+
+        public $typemap(jboxtype, ValueType) getValue() {
+          return iterator.getValue();
+        }
+
+        public $typemap(jboxtype, ValueType) setValue($typemap(jboxtype, ValueType) newValue) {
+          $typemap(jboxtype, ValueType) oldValue = iterator.getValue();
+          iterator.setValue(newValue);
+          return oldValue;
+        }
+      }.init(itr));
+      itr = itr.getNextUnchecked();
+    }
+
+    return setToReturn;
+  }
+%}
+
+  public:
+    map();
+    map(const map<KeyType, ValueType, Comparator >&);
+
+    struct iterator {
```

Comment:
The iterator class needn't be public, so add this inside class definition on the next line:
    %typemap(javaclassmodifiers) iterator "protected class"